api/api_calls.py

```diff
@@ -527,26 +524,28 @@ def _send_file(filename, parameter_name, bytes_read=0):
                    "Content-Disposition: form-data; name=\"{parameter_name}\"; filename=\"{filename}\"\r\n\r\n").format(
                 boundary=boundary, parameter_name=parameter_name, filename=filename.replace("\\", "/"))).encode()
 
+            # Get total file size for progress
+            total_file_size = path.getsize(filename)
+
             # Send the contents of the file, read_size bytes at a time until
             # we've either read the entire file, or we've been instructed to
             # stop the upload by the UI
             logging.info("Starting to send file {}".format(filename))
             try:
                 with open(filename, "rb", read_size) as fastq_file:
                     data = fastq_file.read(read_size)
-                    # Todo: I'm not a big fan of having this here, but until the message passing/threading is done it is a easy way to show progress to the user on CLI
-                    dot_print_factor = 5 * read_size  # todo:remove when message passing/threading
-                    dot_read = 0  # todo:remove when message passing
+                    # Command line progress info printing
+                    # Todo: once message passing is in place, this might find its home in that module
+                    bytes_read = 0
                     while data and not self._stop_upload:
                         bytes_read += len(data)
-                        dot_read +=  len(data)  # todo:remove when message passing
-                        if dot_read % dot_print_factor == 0:  # todo:remove when message passing
-                            print(".", end="")  # this prints dots as the file is uploaded so the user can see progress
+                        print("Progress: ", round(bytes_read/total_file_size*100, 2),
+                              "% Uploaded     \r", end="")
                         # todo: rework send_message
                         # send_message(sample.upload_progress_topic, progress=bytes_read)
```

Comment:
Ditch the commented out message passing stuff until we're ready to go with it please.

Reply:
I removed all the todo's related to message passing too, when we add it the old locations probably won't line up exactly to where new messages will go.